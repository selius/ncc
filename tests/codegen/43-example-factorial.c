int factorial_recursive(int n)
{
	if (n <= 1) {
		return 1;
	}

	return n * factorial_recursive(n - 1);
}

int factorial_loop(int n)
{
	int i;
	int res;
	res = 1;

	for (i = 2; i <= n; i++) {
		res *= i;
	}
	
	return res;
}

int main()
{
	__print_int(factorial_recursive(5));
	__print_int(factorial_recursive(7));

	__print_int(factorial_loop(5));
	__print_int(factorial_loop(7));


	return 0;
}
