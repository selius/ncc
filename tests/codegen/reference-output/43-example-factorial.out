120
5040
120
5040
